### PR TITLE
Don't break existing consumers

### PR DESCRIPTION
**Pull request by richardhopton**

### CODE REVIEW

File: HobknobClientNet/HobknobClientFactory.cs [hunk 1]
```diff
@@ -4,22 +4,21 @@ namespace HobknobClientNet
 {
     public interface IHobknobClientFactory
     {
-        IHobknobClient Create(string etcdHost, int etcdPort, string applicationName, TimeSpan cacheUpdateInterval, EventHandler<CacheUpdateFailedArgs> cacheUpdateFailed);
+        IHobknobClient Create(string etcdHost, int etcdPort, string applicationName, TimeSpan cacheUpdateInterval, EventHandler<CacheUpdateFailedArgs> cacheUpdateFailed = null);
```
Comment (gondar) [inline HobknobClientNet/HobknobClientFactory.cs]: I had chat with original creators of Hobknob asking why in the first place was it throwing if connection to etcd failed. What I groked from it was that it was meant to be opinionated piece of code and thus protect customers from making mistake of not protecting from failure.
To keep that spirit but not enforce any particular way of doing it I wanted to force existing customers to think about error handling. As for backwards compatibility, there is no reason to update existing client except for the case if you want to actually take advantage of new interface and instead of failing do something more meaningful if etcd is not available.
In short I think this change goes against spirit of this library and could result in bad surprise to somebody updating the client. But I don't care that much! :-)
